pkg/k8s/testutils/resources.go

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package testutils
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	discov1 "k8s.io/api/discovery/v1"
+	discov1beta1 "k8s.io/api/discovery/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	cilium_v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+)
+
+var (
+	DefaultVersion = "1.26"
+
+	// APIResources is the list of API resources for the k8s version that we're mocking.
+	// This is mostly relevant for the feature detection at pkg/k8s/version/version.go.
+	// The lists here are currently not exhaustive and expanded on need-by-need basis.
+	APIResources = map[string][]*metav1.APIResourceList{
+		"1.16": {
+			Corev1APIResources,
+			Ciliumv2APIResources,
+		},
+		"1.24": {
+			Corev1APIResources,
+			DiscoveryV1APIResources,
+			DiscoveryV1beta1APIResources,
+			Ciliumv2APIResources,
+		},
+		"1.25": {
+			Corev1APIResources,
+			DiscoveryV1APIResources,
+			Ciliumv2APIResources,
+		},
+		"1.26": {
+			Corev1APIResources,
+			DiscoveryV1APIResources,
+			Ciliumv2APIResources,
+		},
+	}
+
+	Corev1APIResources = &metav1.APIResourceList{
```

Comment:
Nit: rename `Corev1APIResources -> CoreV1APIResources` in line with `DiscoveryV1APIResources`, and similarly `Ciliumv2APIResources -> CiliumV2APIResources`. Super-nit: `DiscoveryV1beta1APIResources -> DiscoveryV1Beta1APIResources`.